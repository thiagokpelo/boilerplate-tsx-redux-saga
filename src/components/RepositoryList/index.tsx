import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ApplicationState } from 'store';
import { Repository } from 'store/ducks/repositories/types';
import * as RepositoriesActions from 'store/ducks/repositories/actions';
import { bindActionCreators, Dispatch } from 'redux';

import RepositoryItem from 'components/RepositoryItem';

interface StateProps {
  repositories: Repository[]
}

interface DispatchProps {
  loadRequest(): void
}

interface OwnProps { }

type Props = StateProps & DispatchProps & OwnProps

class RepositoryList extends Component<Props> {
  componentDidMount() {
    const { loadRequest } = this.props;
    loadRequest();
  }

  render() {
    const { repositories } = this.props;

    return (
      <ul>{repositories.map((repository) => <RepositoryItem key={repository.id} repository={repository} />)}</ul>
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(RepositoriesActions, dispatch);

const mapStateToProps = (state: ApplicationState) => ({ repositories: state.repositories.data });

export default connect(mapStateToProps, mapDispatchToProps)(RepositoryList);
