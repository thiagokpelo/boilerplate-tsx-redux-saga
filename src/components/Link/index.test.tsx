import React from 'react';
import { shallow } from 'enzyme';
import RepositoryList from '.';

describe('Link', () => {
  it('Renders link to Google', () => {
    const link = shallow(<RepositoryList href="http://google.com">Link to Google</RepositoryList>);
    expect(link).toMatchSnapshot();
  });

  it('Renders link to Google with classname', () => {
    const link = shallow(<RepositoryList href="http://google.com" className="my-link-class">Link to Google</RepositoryList>);
    expect(link).toMatchSnapshot();
  });
});
