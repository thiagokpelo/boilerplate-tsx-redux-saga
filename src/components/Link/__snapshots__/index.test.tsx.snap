// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Renders link to Google 1`] = `ShallowWrapper {}`;

exports[`Link Renders link to Google with classname 1`] = `ShallowWrapper {}`;
